### ISSUE TITLE
Obtaining Rights to Custom Code in Contracts

### ISSUE THREAD

**Issue by mgifford** [canada-ca/open-source-logiciel-libre#34]
Wow, we've got a long ways to go with open source & accessibility.  How on earth is this not the other way around.  If any contractor wants to own the IP that should be the exception, not the norm. How are we expecting innovation to happen in Canada if we keep locking up our IP in 3rd party staffing agencies and USA owned companies? 

Just noting this as a major policy hurdle. 

https://github.com/canada-ca/open-source-logiciel-libre/blob/master/en/guides/publishing-open-source-code.md#obtain-rights-to-custom-code-in-contracts

**Comment by gcharest** [comment]
It's a policy hurdle but from the scope of guidelines, with the current policies in place, this is unfortunately required.

Doesn't mean we can't point it out to the the responsible body of that policy that it should be updated in a way that supports the entire market rather than single entities.

**Comment by mgifford** [comment]
Who would be responsible for changing this?  Is this something that goes up to the Minister?

**Comment by gcharest** [comment]
I will be 100% honest, I'm not sure. I do think that would be a political review but most likely lead internally.

We're looking at this policy from a very specific use case and they may not have been as familiar with the impact to our ability to release open source software at the moment of developing it.

**Comment by mgifford** [comment]
Thanks @gcharest there's still so much work to do.

**Comment by vzvenyach** [comment]
With the opening disclaimer that I am no expert in Canadian procurement policy, as I read the existing implementation guidelines, there are some robust opportunities for requiring contractors to deliver their work under a free and open source license.

1. *Foreground IP when dealing with legacy systems*. To the extent that the Crown already has ownership or license of a legacy system, and a contractor develops new foreground IP in the modification/improvement of the existing legacy system, the [existing policies](https://www.ic.gc.ca/eic/site/068.nsf/eng/h_00001.html#p4.2) specify that the Crown ought to "retain[] the ownership on the legacy system as well as any newly created Foreground IP." In such situations, the Crown may choose to open-source the existing codebase and require that Foreground IP be licensed under the same terms to promote government and commercial reuse.

2. *When multiple vendors are involved*. Similarly, the [existing policies](https://www.ic.gc.ca/eic/site/068.nsf/eng/h_00001.html#p4.3) provide that if two or more vendors are developing components of a system, the Crown shall "require[] IP ownership of the deliverables (from both contractors) in order to incorporate each deliverable into a complete system as a prerequisite to the planned transfer to the private sector for the purposes of commercial exploitation." Here, too, providing for FOSS licenses may be advantageous.

3. *Where open-source licenses allow for commercialization*. It is important to note that in the event the Foreground IP is licensed under an open-source license, the contractor still may have numerous opportunities to commercialize that IP. For example, there is no _obligation_ on the contractor to license future versions or non-foreground IP features of the source code under an open-source license (i.e., use an "open core model"). Similarly, a copyleft license could allow for paid licenses by the contractor (i.e., "copyleft plus paid mode"). There are also numerous open-source companies that provide the software as a SaaS offering as well. In other words, using an open-source license meets the Crown's policy of enabling the contractor to own and commercialize Foreground IP while satisfying government's interests in reuse and broad commercial availability and future competition.

4. *Where there is a high likelihood of government-contractor collaboration*. In my experience at 18F, there were many situations where government employees and non-government employees collaborated on the work. In such situations, we found it preferable to remove authorship issues and simply declare the source code to be part of the public domain. This removed the friction associated with public-private collaboration.

Finally, I'd note that standard clause K3030C ("the Clause") does not _conflict_ with a substantive requirement for FOSS licensing. For example, if the government included the Clause and the vendor selected a copyleft license for Foregound IP, the government could still theoretically choose to "fork" the code and keep the fork non-source-disclosed because the Crown maintains the right to do so under the Clause. A member of the public, by contrast, could not do so without a separate licensing agreement with the contractor. Accordingly, even if the Crown used the Clause in a contract, the government could still choose to require a contractor to publish their software under a free and open source license as part of the deliverables.

**Comment by gcharest** [comment]
Thanks for this well written observation!

I'll have to have a second read when at my computer but I think we can most
likely include those in the guidelines!

Le jeu. 9 mai 2019 18 h 56, V David Zvenyach <notifications@github.com> a
écrit :
